src/App.js

```diff
@@ -1,26 +1,126 @@
 import React from 'react';
-import logo from './logo.svg';
+import FoodBox from './components/FoodBox';
+import AddFood from './components/AddFood';
+import Search from './components/Search';
+
+import foods from './foods.json';
+
+import 'bulma/css/bulma.css';
 import './App.css';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+class App extends React.Component {
+  state = {
+    cloneFoods: foods,
+    addForm: false,
+    todaysFoods: [],
+    totalCal: 0,
+  };
+
+  // Form handler //
+  handleAdd = (e) => {
+    e.preventDefault();
+    let name = e.target.addFoodName.value;
+    let image = e.target.addFoodImage.value;
+    let calories = e.target.addFoodCalories.value;
+
+    let newFood = {
+      name: name,
+      image: image,
+      calories: calories,
+      quantity: 0,
+    };
+    this.setState({
+      cloneFoods: [...foods, newFood],
+      addForm: false,
+    });
+  };
+
+  handleForm = () => {
+    this.setState({
+      addForm: true,
+    });
+  };
+
+  // Search handler //
+  handleSearch = (e) => {
+    e.preventDefault();
+    const { cloneFoods } = this.state;
+    let searchInput = e.target.value.toLowerCase();
+    let results = cloneFoods.filter((food) => {
```

Comment:
the cloneFoods state will continue to reduce everytime you change the search input, meaning it won't include the original data and if you try to delete the input in won't find any matches from the original data. 
To solve this quick you can use the original data variable **foods.filter** instead of the state **cloneFoods.filter**
A better solution though, would be to have a separate state that holds the original data and will never be altered, that way you can access this info from the state when you need all the info and not something that was changed along the way.